nsqd/nsqd.go

```diff
@@ -298,6 +324,12 @@ func (n *NSQD) LoadMetadata() {
 }
 
 func (n *NSQD) PersistMetadata() error {
+	if n.getFlag(flagLoading) {
```

Comment:
since notification (and persisting) happens asynchronously, I think we need to ensure that we don't even _try_ to notify so there isn't any race - i.e. do this logic where the notification is sent rather than received.